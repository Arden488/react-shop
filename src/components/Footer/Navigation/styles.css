.footer-navigation {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-column-gap: 10px;
  grid-row-gap: 5px;
}

.footer-navigation__item {
  color: #666;
  font-size: 13px;
  text-decoration: none;
}
