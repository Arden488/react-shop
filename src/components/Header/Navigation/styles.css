.header-navigation {
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 10px;
  align-items: center;
  margin-top: -3px; /* capital letters compensation */
  padding-right: 100px
}

.header-navigation__item {
  font-size: 14px;
  text-decoration: none;
  color: #fff;
}

.header-navigation__item_active {
  text-decoration: underline;
}
