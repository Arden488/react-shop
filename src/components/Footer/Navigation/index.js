import React from 'react';
import { Link } from 'react-router-dom';

import mainNavItems from '../../../config/mainNav';

import './styles.css';

const Navigation = () => (
  <nav className="footer-navigation">
    {mainNavItems.map(item => <Link className="footer-navigation__item" key={item.path} to={item.path}>{item.title}</Link>)}
  </nav>
);

export default Navigation;
