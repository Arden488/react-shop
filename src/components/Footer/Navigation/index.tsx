import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import mainNavItems from '../../../config/mainNav';

import './styles.css';

class Navigation extends Component {
  renderLink(item) {
    let linkClass = 'footer-navigation__item';
    if (this.props.location.pathname === item.path) {
      linkClass += ' footer-navigation__item_active';
    }
    return (
      <Link
        className={linkClass}
        key={item.path}
        to={item.path}
      >
        {item.title}
      </Link>
    );
  }

  render() {
    return (
      <nav className="footer-navigation">
        {mainNavItems.map(item => this.renderLink(item))}
      </nav>
    );
  }
}

Navigation.propTypes = {
  location: PropTypes.shape().isRequired,
};

export default withRouter(Navigation);
