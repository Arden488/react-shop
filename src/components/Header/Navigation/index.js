import React from 'react';
import { Link } from 'react-router-dom';

import mainNavItems from '../../../config/mainNav';

import './styles.css';

const Navigation = () => (
  <nav className="header-navigation">
    {mainNavItems.map(item => <Link className="header-navigation__item" key={item.path} to={item.path}>{item.title}</Link>)}
  </nav>
);

export default Navigation;
