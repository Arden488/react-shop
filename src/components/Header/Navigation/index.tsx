import * as React from 'react';
import { Link, RouteComponentProps, withRouter } from 'react-router-dom';

import mainNavItems from '../../../config/mainNav';

import './styles.css';

interface INavigationProps extends RouteComponentProps<any> {
  location: any,
}

class Navigation extends React.Component<INavigationProps> {
  public render() {
    return (
      <nav className="header-navigation">
        {mainNavItems.map(item => this.renderLink(item))}
      </nav>
    );
  }

  public renderLink(item: { path: string, title: string }) {
    let linkClass = 'header-navigation__item';
    if (this.props.location.pathname === item.path) {
      linkClass += ' header-navigation__item_active';
    }
    return (
      <Link
        className={linkClass}
        key={item.path}
        to={item.path}
      >
        {item.title}
      </Link>
    );
  }
}

export default withRouter(Navigation);
