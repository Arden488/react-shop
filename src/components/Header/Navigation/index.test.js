/* global it, expect, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';

import { Link } from 'react-router-dom';
import Navigation from './index';

let component = null;

beforeEach(() => {
  component = shallow(<Navigation />);
});

it('renders without crashing', () => {
  expect(component.exists()).toEqual(true);
});

it('should have Links', () => {
  expect(component.find(Link).length).toBeGreaterThan(1);
});
