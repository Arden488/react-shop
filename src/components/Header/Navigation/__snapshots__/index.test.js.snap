// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="header-navigation"
>
  <Link
    className="header-navigation__item header-navigation__item_active"
    key="/"
    replace={false}
    to="/"
  >
    <a
      className="header-navigation__item header-navigation__item_active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/catalog"
    replace={false}
    to="/catalog"
  >
    <a
      className="header-navigation__item"
      href="/catalog"
      onClick={[Function]}
    >
      Catalog
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/catalog/iphone"
    replace={false}
    to="/catalog/iphone"
  >
    <a
      className="header-navigation__item"
      href="/catalog/iphone"
      onClick={[Function]}
    >
      iPhone
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/catalog/ipad"
    replace={false}
    to="/catalog/ipad"
  >
    <a
      className="header-navigation__item"
      href="/catalog/ipad"
      onClick={[Function]}
    >
      iPad
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/catalog/mac"
    replace={false}
    to="/catalog/mac"
  >
    <a
      className="header-navigation__item"
      href="/catalog/mac"
      onClick={[Function]}
    >
      Mac
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/catalog/accessories"
    replace={false}
    to="/catalog/accessories"
  >
    <a
      className="header-navigation__item"
      href="/catalog/accessories"
      onClick={[Function]}
    >
      Accessories
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/news"
    replace={false}
    to="/news"
  >
    <a
      className="header-navigation__item"
      href="/news"
      onClick={[Function]}
    >
      News and events
    </a>
  </Link>
  <Link
    className="header-navigation__item"
    key="/payment-and-delivery"
    replace={false}
    to="/payment-and-delivery"
  >
    <a
      className="header-navigation__item"
      href="/payment-and-delivery"
      onClick={[Function]}
    >
      Delivery and payment
    </a>
  </Link>
</nav>
`;
